File: Binary Search Tree Maven version/src/main/java/model/constants/MessageTexts.java
package model.constants;

// TODO: Make this class read the required phrases from config .json, .xml or .lua file instead of storing them as literals
public final class MessageTexts {

    // Main menu
    public static final String initialMenuText = """
Welcome to TextParser!
Please, state where do you want to get the text from:
1. Local file
2. File in the web
3. Web Page contents
4. Web Page contents (Raw)
""";

    // Additional data request
    public static final String localFilePathRequestText = "Please, state the path to the text file.";

    public static final String webPageUrlAddressRequestText = "Please, state the URL of the source website.";

    public static final String soughtForTokenRequestText = "State the sought-for word.";


    // Exceptions Notifications
    public static final String invalidInputNotificationText = "Unfortunately, we cannot interpret your input.\n" +
                                                                "Please, check its consistency and try again.";

    public static final String readingSessionFailureNotificationText = "Attempt to receive data from the given source resulted in a failure.\n" +
                                                                "Please, try again later!";

    public static final String unknownErrorNotificationText = "Something went wrong whilst processing your request.\n" +
                                                                "Please, check the consistency of your input or try again later!";

}

File: src/main/java/model/constants/message_texts.lua
return
{
    -- Data request messages
    request_message_beginning = "Please, state the ",

    local_file_path_request = request_message_beginning .. "path to the text file.",
    web_page_url_request = request_message_beginning .. "URL of the source website.",
    sought_for_token_request = request_message_beginning .. "sought-for word.",

    -- Error notifications
    error_message_beginning = "An error occurred!\n",

    invalid_input_notification = error_message_beginning .. "Unfortunately, we cannot interpret your input.\nPlease, check its consistency and try again.",
    reading_session_failure_notification = error_message_beginning .. "Attempt to receive data from the given source resulted in a failure.\nPlease, try again later!",
    unknown_error_notification = error_message_beginning .. "Something went wrong whilst processing your request.\nPlease, check the consistency of your input or try again later!"
}